api/metrics/views.py

```diff
@@ -154,3 +160,42 @@ def metric_type(self):
     @property
     def metric(self):
         return PreprintDownload
+
+class RawMetricsView(GenericAPIView):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        IsRawMetricsUser,
+        TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.METRICS_BASIC]
+    required_write_scopes = [CoreScopes.METRICS_RESTRICTED]
+
+    view_category = 'raw-metrics'
+    view_name = 'raw-metrics-view'
+
+    serializer_class = RawMetricsSerializer
+
+    @require_switch(ENABLE_RAW_METRICS)
```

Comment:
No need to change it, but another way to do this would have been to add a permissions class to check for the switch.